README.md

# <img alt="rocketsled" src="docs_rst/_static/logo-small.png" width="350">

| Tests  |   Code Coverage   |  Codacy | Release |
|:----------:|:-------------:|:------:|:------:|
[![CircleCI](https://img.shields.io/circleci/project/github/hackingmaterials/rocketsled/master.svg)](https://circleci.com/gh/hackingmaterials/rocketsled) | [![Codacy Badge](https://img.shields.io/codacy/coverage/3e4e2ac81f8d47c58759c386a6377e7d.svg?colorB=brightgreen)](https://www.codacy.com/app/ardunn/rocketsled) | [![Codacy Badge](https://img.shields.io/codacy/grade/3e4e2ac81f8d47c58759c386a6377e7d.svg)](https://www.codacy.com/app/ardunn/rocketsled) | [![PyPI version](https://img.shields.io/pypi/v/rocketsled.svg?colorB=blue)](https://pypi.org/project/rocketsled/) |


rocketsled is a black-box optimization framework "on rails" for high-throughput computation with FireWorks.

- **Website (including documentation):** https://hackingmaterials.github.io/rocketsled/
- **Help/Support:** https://discuss.matsci.org
- **Source:** https://github.com/hackingmaterials/rocketsled
- **FireWorks website:** https://materialsproject.github.io/fireworks

If you find rocketsled useful, please encourage its development by citing the [following paper](http://doi.org//10.1088/2515-7639/ab0c3d) in your research:

```
Dunn, A., Brenneck, J., Jain, A., Rocketsled: a software library for optimizing
high-throughput computational searches. J. Phys. Mater. 2, 034002 (2019).
```

If you find FireWorks useful, please consider citing [its paper](http://dx.doi.org/10.1002/cpe.3505) as well:

```
Jain, A., Ong, S. P., Chen, W., Medasani, B., Qu, X., Kocher, M., Brafman, M., 
Petretto, G., Rignanese, G.-M., Hautier, G., Gunter, D., and Persson, K. A. 
FireWorks: a dynamic workflow system designed for high-throughput applications.
Concurrency Computat.: Pract. Exper., 27: 5037–5059. (2015)
```


CHANGELOG.md
# Changelog

## [Unreleased](https://github.com/hackingmaterials/rocketsled/tree/HEAD)

[Full Changelog](https://github.com/hackingmaterials/rocketsled/compare/v2019.9.12...HEAD)

**Implemented enhancements:**

- Change versioning [\#81](https://github.com/hackingmaterials/rocketsled/issues/81)
- Enforce a code style [\#78](https://github.com/hackingmaterials/rocketsled/issues/78)

**Fixed bugs:**

- Prepend hostname to the PID for queue checking [\#80](https://github.com/hackingmaterials/rocketsled/issues/80)
- Fix codacy issues [\#58](https://github.com/hackingmaterials/rocketsled/issues/58)
- "scaling" portion of code is confusing [\#53](https://github.com/hackingmaterials/rocketsled/issues/53)

**Closed issues:**

- Update help link to new forum [\#82](https://github.com/hackingmaterials/rocketsled/issues/82)
- upgrade pymongo version [\#72](https://github.com/hackingmaterials/rocketsled/issues/72)

**Merged pull requests:**

- fix some codacy issues [\#92](https://github.com/hackingmaterials/rocketsled/pull/92) ([ardunn](https://github.com/ardunn))
- formatting fixes [\#91](https://github.com/hackingmaterials/rocketsled/pull/91) ([ardunn](https://github.com/ardunn))
- fix req conflict [\#90](https://github.com/hackingmaterials/rocketsled/pull/90) ([ardunn](https://github.com/ardunn))
- Bump matplotlib from 3.1.1 to 3.2.1 [\#89](https://github.com/hackingmaterials/rocketsled/pull/89) ([dependabot-preview[bot]](https://github.com/apps/dependabot-preview))
- Pidhostame [\#88](https://github.com/hackingmaterials/rocketsled/pull/88) ([ardunn](https://github.com/ardunn))
- Bump scipy from 1.3.1 to 1.4.1 [\#87](https://github.com/hackingmaterials/rocketsled/pull/87) ([dependabot-preview[bot]](https://github.com/apps/dependabot-preview))
- Bump fireworks from 1.9.4 to 1.9.5 [\#86](https://github.com/hackingmaterials/rocketsled/pull/86) ([dependabot-preview[bot]](https://github.com/apps/dependabot-preview))
- Bump pymongo from 3.9.0 to 3.10.1 [\#84](https://github.com/hackingmaterials/rocketsled/pull/84) ([dependabot-preview[bot]](https://github.com/apps/dependabot-preview))
- Bump numpy from 1.16.3 to 1.17.2 [\#79](https://github.com/hackingmaterials/rocketsled/pull/79) ([dependabot-preview[bot]](https://github.com/apps/dependabot-preview))
- Bump scipy from 1.2.1 to 1.3.1 [\#77](https://github.com/hackingmaterials/rocketsled/pull/77) ([dependabot-preview[bot]](https://github.com/apps/dependabot-preview))
- Bump fireworks from 1.9.0 to 1.9.4 [\#76](https://github.com/hackingmaterials/rocketsled/pull/76) ([dependabot-preview[bot]](https://github.com/apps/dependabot-preview))
- Bump matplotlib from 3.0.3 to 3.1.1 [\#75](https://github.com/hackingmaterials/rocketsled/pull/75) ([dependabot-preview[bot]](https://github.com/apps/dependabot-preview))
- Bump pymongo from 3.7.2 to 3.9.0 [\#74](https://github.com/hackingmaterials/rocketsled/pull/74) ([dependabot-preview[bot]](https://github.com/apps/dependabot-preview))
- Bump scikit-learn from 0.20.3 to 0.21.3 [\#73](https://github.com/hackingmaterials/rocketsled/pull/73) ([dependabot-preview[bot]](https://github.com/apps/dependabot-preview))

## [v2019.9.12](https://github.com/hackingmaterials/rocketsled/tree/v2019.9.12) (2019-09-11)

[Full Changelog](https://github.com/hackingmaterials/rocketsled/compare/v2019.9.11...v2019.9.12)

## [v2019.9.11](https://github.com/hackingmaterials/rocketsled/tree/v2019.9.11) (2019-09-11)

[Full Changelog](https://github.com/hackingmaterials/rocketsled/compare/v2019.6.5...v2019.9.11)

**Closed issues:**

- General comment on code and examples [\#56](https://github.com/hackingmaterials/rocketsled/issues/56)

## [v2019.6.5](https://github.com/hackingmaterials/rocketsled/tree/v2019.6.5) (2019-06-05)

[Full Changelog](https://github.com/hackingmaterials/rocketsled/compare/v2019.6.4...v2019.6.5)

## [v2019.6.4](https://github.com/hackingmaterials/rocketsled/tree/v2019.6.4) (2019-06-05)

[Full Changelog](https://github.com/hackingmaterials/rocketsled/compare/v2019.5.3...v2019.6.4)

## [v2019.5.3](https://github.com/hackingmaterials/rocketsled/tree/v2019.5.3) (2019-05-04)

[Full Changelog](https://github.com/hackingmaterials/rocketsled/compare/v2019.5.2...v2019.5.3)

## [v2019.5.2](https://github.com/hackingmaterials/rocketsled/tree/v2019.5.2) (2019-05-04)

[Full Changelog](https://github.com/hackingmaterials/rocketsled/compare/v2019.4.17...v2019.5.2)

## [v2019.4.17](https://github.com/hackingmaterials/rocketsled/tree/v2019.4.17) (2019-04-18)

[Full Changelog](https://github.com/hackingmaterials/rocketsled/compare/v2019.3.15...v2019.4.17)

**Closed issues:**

- add paper citation to rocketsled [\#42](https://github.com/hackingmaterials/rocketsled/issues/42)

## [v2019.3.15](https://github.com/hackingmaterials/rocketsled/tree/v2019.3.15) (2019-03-16)

[Full Changelog](https://github.com/hackingmaterials/rocketsled/compare/v2019.2.27...v2019.3.15)

**Closed issues:**

- Transition to unittest [\#64](https://github.com/hackingmaterials/rocketsled/issues/64)
- Comprehensive guide needs update [\#59](https://github.com/hackingmaterials/rocketsled/issues/59)
- tutorial rework [\#55](https://github.com/hackingmaterials/rocketsled/issues/55)

## [v2019.2.27](https://github.com/hackingmaterials/rocketsled/tree/v2019.2.27) (2019-02-27)

[Full Changelog](https://github.com/hackingmaterials/rocketsled/compare/v2019.2.17...v2019.2.27)

**Merged pull requests:**

- Fix typo, when all inputs are floats [\#65](https://github.com/hackingmaterials/rocketsled/pull/65) ([RemiLehe](https://github.com/RemiLehe))

## [v2019.2.17](https://github.com/hackingmaterials/rocketsled/tree/v2019.2.17) (2019-02-19)

[Full Changelog](https://github.com/hackingmaterials/rocketsled/compare/v2019.1.1...v2019.2.17)

**Merged pull requests:**

- Minor corrections to the tutorial and examples [\#63](https://github.com/hackingmaterials/rocketsled/pull/63) ([RemiLehe](https://github.com/RemiLehe))


old/reference.py
'''
References for unit tests and development.
'''

__author__ = "Alexander Dunn <ardunn@lbl.gov>"
__version__ = 0.1




simple_dict = {'A':14,
               'B':12,
               'C':91.1}

cars_dict = {
    'Car':{
        'Toyota':{
            'Corolla': 42,
            'Tundra': 27,
        },
        'Ford':{
            'F250':19
        }
    },
    'Pet':{
        'Dog': 5,
        'Cat':8
    }
}

ref_dict = {'Structure':{
                    'A':1.31,
                    'B':1.03,
                    'C':2.55},
                'Group':{
                    'g1': 100,
                    'g2': 200,
                    'g3': 400,
                },
                'e_above_hull': 0.001,
                'types': {
                    'old':{
                        'q': 88,
                        'r': 99
                    },
                    'new':{
                        's': 65,
                        't': 53
                    }}}

ref_dict2 = {'Structure':{
                    'A':2.21,
                    'B':2.03,
                    'C':3.55},
                'Group':{
                    'g1': 300,
                    'g2': 300,
                    'g3': 700,
                },
                'e_above_hull': 0.008,
                'types': {
                    'old':{
                        'q': 22,
                        'r': 11
                    },
                    'new':{
                        's': 29,
                        't': 21
                    }}}
